avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,255 @@
+@description('Required. The name of the Azure Container Registry.')
+param acrName string
+
+@description('Optional. How the deployment script should be forced to execute. Default is to force the script to deploy the image to run every time.')
+param forceUpdateTag string = utcNow()
+
+@description('Optional. Azure RoleId that are required for the DeploymentScript resource to import images. Default is AcrPush, which is needed to import into an ACR.')
+// param rbacRoleNeeded string = '8311e382-0749-4cb8-b61a-304f252e45ec' // AcrPush
+param rbacRoleNeeded string = 'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor is needed to import ACR
+
+@description('Optional. Does the Managed Identity already exists, or should be created. Default is false.')
+param useExistingManagedIdentity bool = false
+
+@description('Conditional. Name of the Managed Identity resource to create. Required if `useExistingManagedIdentity` is `true`. Defaults to `id-ContainerRegistryImport`.')
+param managedIdentityName string = 'id-ContainerRegistryImport'
+
+@description('Conditional. For an existing Managed Identity, the Subscription Id it is located in. Default is the current subscription. Required if `useExistingManagedIdentity` is `true`. Defaults to the curent subscription.')
+param existingManagedIdentitySubId string = subscription().subscriptionId
+
+@description('Conditional. For an existing Managed Identity, the Resource Group it is located in. Default is the current resource group. Required if `useExistingManagedIdentity` is `true`. Defaults to the current resource group.')
+param existingManagedIdentityResourceGroupName string = resourceGroup().name
+
+@description('Required. A fully qualified image name to import.')
+@metadata({
+  example: 'mcr.microsoft.com/k8se/quickstart-jobs:latest'
+})
+param image string
+
+@description('Optional. The image will be overwritten if it already exists in the ACR with the same tag. Default is false.')
+param overwriteExistingImage bool = false
+
+@description('Optional. A delay in seconds before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate. Default is 30s.')
+param initialScriptDelay int = 30
+
+@description('Optional. The maximum number of retries for the script import operation. Default is 3.')
+param retryMax int = 3
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('Optional. When the script resource is cleaned up. Default is OnExpiration and the cleanup time is after 1h.')
+param cleanupPreference string = 'OnExpiration'
+
+@description('Optional. The name of the storage account to use for the deployment script. An existing storage account is needed, if PrivateLink is going to be used for the deployment script.')
+param storageAccountName string = ''
+
+@description('Optional. The subnet id to use for the deployment script. An existing subnet is needed, if PrivateLink is going to be used for the deployment script.')
+param subnetId string = ''
```

Comment:
In alignment with the specs this should be `subnetResourceId`.

Also mind you, it can be multiple subnet resource Ids. Ref:

https://github.com/Azure/bicep-registry-modules/blob/a948d9736be0c5775cde9fb3dc8bc14388e41803/avm/res/resources/deployment-script/main.bicep#L47